gslb/ingestion/fullsync.go

```diff
@@ -399,14 +399,16 @@ func GenerateModels(gsCache *avicache.AviCache) {
 			continue
 		}
 		tenant, hmName := hmKey.Tenant, hmKey.Name
-		gsName, err := avicache.GetGSFromHmName(hmName)
+		gsName, gen, err := avicache.GetGSFromHmName(hmName)
 		if err != nil {
-			gslbutils.Debugf("key: %v, msg: can't get gs name from hm, err : %v", hmKey, err)
-			continue
+			gslbutils.Logf("key: %v, msg: can't get gs name from hm, err : %v", hmKey, err)
+			if !gslbutils.HMCreatedByAMKO(hmName) || hmName == gslbutils.SystemGslbHealthMonitorPassthrough {
+				continue
+			}
 		}
 		gsKey := tenant + "/" + gsName
 		found, _ := agl.Get(gsKey)
-		if found {
+		if found && gen == 1 {
```

Comment:
Ideally instead of returning 3 variables from the same method, I would just handle `gen == 1` inside the `GetGSFromHmName` itself.

Reply:
The variable `gen` is being returned here instead of just handling it in `GetGSFromHmName` as we need to see if gsKey exists in the store or not. Only if the key exists and the HM is a 1st gen/generation HM, we continue.

Reply:
@sudswasavi  It can be performed in the `GetGSFromHmName` function too. We have 2 options
1. We'll have to move code from line 403 to 415 in the function `GetGSFromHmName`. But there's a function separation here. Making this change would mean, performing the ingestion layer tasks in a cache function, which might not be a good idea.
2. We can move line 403 to 408. Which would mean that, we'll be returning 3 variables again - `gsName`, `gsKey`, and lets say another parameter named `action` that we need to decide if you want to `continue` from the `for` loop or not
cc @hemantavi